docs/enhancements/agent-upgrade.md

```diff
@@ -0,0 +1,189 @@
+---
+title: agent-upgrade
+authors:
+  - "@jhernand"
+creation-date: 2022-06-13
+last-updated: 2022-06-13
+---
+
+# Agent upgrade
+
+## Summary
+
+Usually the time since the agent is started and the cluster is installed is
+short. But in some situations it may be long, even many days, and during that
+long time the service may be upgraded and require a newer version of the agent.
+The current mechanism to perform that agent upgrade is to reboot the host. That
+is inconvenient, specially if there are many hosts or if rebooting a host
+requires manual intervention. This is a proposal to implement a mechanism to
+upgrade the agent without having to reboot the host.
+
+## Motivation
+
+Reduce the need to reboot hosts when there are changes in the service that
+require new versions of the agent.
+
+### Goals
+
+A host that has been booted and registered with matching versions of the agent
+and the service will not need to be rebooted in order to install the cluster
+with a newer version of the service.
```

Comment:
```suggestion
with a newer version of the service.
```

This is not really about newer versions of the service, but more about service pods which expect a newer version of the agent. The service version itself is not a concern here

Reply:
Done.